File: Embedded_C_for_Spectrum_Digital_eZdsp_28335/processLSpike3.c
/*
 * processLSpike3.c
 *
 *  Created on: Apr 11, 2010
 *      Author: Jeff
 */
#include "CableMonitoring.h"
 
void generateSpectra();
void labelMaxFreqs();
void spike3FindPeakOffsets();


void processLSpike3(){
	float binSpacing;
	int i;
	    
	generateSpectra();

	labelMaxFreqs();

	spike3FindPeakOffsets();

	binSpacing = ((float) SAMPLING_HZ)/((float) DATA_LENGTH);

	// Normalize peaks to freqs
	for (i=0;i<50;i++){
		peakList[i] *= binSpacing;
	}
}

File: Processing_M-Files/processLSpike3.m
% [freqList, freqConsistency] = list of freqs and corr. consistencies
% Based on setFreqrRamp, but simplifying for conversion to C for DSP

% adapted to apply 3 point equilateral triangle fit peak, instead of
% averaged peak. Uses one spectra from beginning of array A to length
% 'dataLength'.

%#eml

function [peakList] = processLSpike3(A,dataLength) 

%define constants
samplingHz = 100; % Data Sampling Rate (100 for design, 160 for scope)

data1 = A(1:dataLength); % change to data input
spectra1 = generateSpectra(data1,samplingHz);
maxFreq1 = labelMaxFreqs(spectra1);
% make list of peaks found using 3 point 2nd order curve fit
%   in terms of relative bin position.
%   normalize to frequency later.
peakOffsetList = spike3FindPeakOffsets(dataLength/2,spectra1,maxFreq1);

% Bin spacing
binSpacing = samplingHz/dataLength;

% Normalize peaks to freqs
peakList = peakOffsetList.*binSpacing;

end